### DISCUSSION TITLE
Rule not firing even if in all.rules

### DISCUSSION THREAD

**Discussion by cac0ns3c** [Security-Onion-Solutions/securityonion#5530] [Unsupported Versions]
Hi Everyone,

I updated my rule set from integration with MISP, everything went fine but the rule is not firing.

Rule:
`cat /opt/so/rules/nids/all.rules | grep 588pat.ryan.ruthepstein.co.uk
alert http $HOME_NET any -> $EXTERNAL_NET $HTTP_PORTS (msg: "MISP e231 [osint:source-type="block-or-filter-list"] Outgoing URL http|3a|//588pat.ryan.ruthepstein.co.uk/"; flow:to_server,established; http.header; content:"588pat.ryan.ruthepstein.co.uk"; fast_pattern; nocase; http.uri; content:"/"; nocase; tag:session,600,seconds; classtype:trojan-activity; sid:114230982; rev:1; priority:4; reference:url,https://*****/events/view/231;)
`
Zeek http logs:
`cat /nsm/zeek/logs/current/http.log | grep 588pat.ryan.ruthepstein.co.uk
{"ts":"2021-09-15T19:13:38.049259Z","uid":"C14RPPzJRfA7EEFAb","id.orig_h":"*****","id.orig_p":1025,"id.resp_h":"91.103.219.223","id.resp_p":80,"trans_depth":1,"method":"GET","host":"588pat.ryan.ruthepstein.co.uk","uri":"/","version":"1.1","user_agent":"*****","request_body_len":0,"response_body_len":157,"status_code":200,"status_msg":"OK","tags":[],"resp_fuids":["FCCvb13Ibk24ftw9Jg"],"resp_mime_types":["text/plain"]}`

What am i missing here?

**Comment by dougburks** [comment]
Are there any clues in the Suricata diagnostic log?
https://docs.securityonion.net/en/2.3/suricata.html#diagnostic-logging

**Comment by cac0ns3c** [comment]
Checked that, there is nothing indicating that the rule is broken or
something like that

**Comment by dougburks** [comment]
Are other alerts firing?

**Reply by cac0ns3c** [reply]
Yeah, this is just an example, all the rules that i have imported from MISP isn't firing, i tried create the event manually ( e.g DNS request via dig in order to fire DNS requests alerts )

**Comment by cac0ns3c** [comment]
Example for a rule from all.rules:
`alert dns any any -> any any (msg: "MISP e225 [osint:source-type="block-or-filter-list"] Domain adobeupdater.ng"; dns.query; content:"adobeupdater.ng"; nocase; pcre: "/(^|[^A-Za-z0-9-])adobeupdater\.ng$/i";  classtype:trojan-activity; sid:7133342; rev:1; priority:4; reference:url,https://****/events/view/225;)`

The DNS query is shown in `/nsm/zeek/logs/current/dns.log`
`{"ts":"2021-09-20T08:05:01.613902Z","uid":"CNgmcR1Mr0OK8FHTQ","id.orig_h":"*****","id.orig_p":52762,"id.resp_h":"8.8.8.8","id.resp_p":53,"proto":"udp","trans_id":19218,"query":"adobeupdater.ng","qclass":1,"qclass_name":"C_INTERNET","qtype":1,"qtype_name":"A","rcode":3,"rcode_name":"NXDOMAIN","AA":false,"TC":false,"RD":true,"RA":false,"Z":0,"rejected":false}`

But nothing in `/nsm/suricata/eve-*.json`

Please help me understand why all the MISP rules aren't firing?

P.S the all.rules file in 1.7Gb if that matters

**Reply by dougburks** [reply]
`all.rules` is 1.7 Gigabytes? That sounds excessive. Have you tried disabling all MISP rules and only enabling one MISP rule as a test?

**Reply by cac0ns3c** [reply]
Tried to run the test with only one rule from local.rules, still no success.
No errors in `/opt/so/log/suricata/suricata.log`

**Comment by clairmont32** [comment]
Have you tried doing a config test with Suricata to ensure it's able to fully load with the current config and all the rules loaded into memory? It'll flag for any errors or warnings that the rules may generate.

`sudo suricata -T -c <path-to-config>`

**Reply by cac0ns3c** [reply]
```
sudo suricata -T -c
sudo: suricata: command not found
```

**Comment by clairmont32** [comment]
Doh! Sorry, my mistake... This would likely need to run inside the
container now. Let me dig up what the equivalent command is.

**Reply by clairmont32** [reply]
`docker exec so-suricata /opt/suricata/bin/suricata -T` or if you have a pcap of the traffic (maybe you can pull it from the pcap page?) `so-suricata-testrule <CustomRule> <TargetPCAP>`

**Reply by cac0ns3c** [reply]
Tried to do it with one rule that passed the `so-suricata-testrule` and it worked.
So the problem sits in the size of the rules.all (1.7Gb)?

**Reply by clairmont32** [reply]
Very plausible. It might be worth working your way down from only critical/high rated events or IOCs. Keep in mind that the more you load in, the longer the packets take to be processed and you may end up hitting a memory buffer limit and not be able to process everything.

**Comment by cac0ns3c** [comment]
Tried to enable only DNS rules, the file is now 100Mb, and still no firing.
/nsm/zeek/logs/current/dns.log has the request
/nsm/suricata/ doesn't have to alert

**Reply by clairmont32** [reply]
Is this still an issue? If so, does the `docker exec` command produce any errors?

**Comment by cac0ns3c** [comment]
Nothing that related to the Suricata rules
